contrib/README.md

```diff
@@ -0,0 +1,137 @@
+# Community Driven Docker Examples
+These docker containers are meant to serve as an isolated develop/deployment environment. Each docker container has the 
+terraform libvirt provider built and placed in the custom plugins folder. The most common use case for these containers 
+is to run your terraform environment in a isolate container talking to a `remote` libvirt system. 
+
+Please refer to the distro's `README.md` for specific instructions.
+
+Additionally, you will find a `Build` folder along side the distro folders. The docker containers within the `Build`
+folder will compile the terraform libvirt provider for you. Please refer to the `Build` folder's `README.md` for more 
+information and instructions.
+
+## Table of Content
+- [Alpine Containers](Alpine/)
+- [Build Containers](Build/)
+- [Debian Conainers](Debian/)
+- [openSUSE Containers](openSUSE/)
+
+
+## General Usage
+There are two types of containers you'll find for each distro, `All-in-One` and `Build-Dependent`.
+
+### All-in-One v.s Build-Dependent Containers
+The `All-in-One` container is a single Dockerfile that you can build and run. The Dockerfile takes advantage of 
+Docker's multi-stage build functionality. It allows you to build a binary or object in a "sub container" and then 
+access the binary or object in your "main container". This Dockerfile is useful if you want to get up and running 
+quickly, but might lead to confusion down the road if plan on using multiple versions of the terraform libvirt provider.
+
+The `Build-Dependent` container relies on another Dockerfile which contains the build of the plugin. To use this 
+container you need to first build the appropriate build Dockerfile and tag it correctly. Once that is done, you are
+able to reference it inside the `Build-Dependent`'s Dockerfile. This design is useful when you are dealing with 
+multiple versions of the terraform libvirt provider such as a stable build, `0.5.2`, which works with Terraform `0.11.X` 
+and the less stable branch, `master`, which currently works with Terraform `0.12.x`.
+
+### Build Args
+There are a couple build args to be aware of when building these various containers.
+
```

Comment:
as general tip for documentation, if you are explaining `args` or `variables` make a list of them so you can enumerate it first with their spec, and the end provide examples.

For this I think it would be better if you create a list of `build arg`, specification and at the end examples.

Take inspiration from https://github.com/dmacvicar/terraform-provider-libvirt/blob/master/website/docs/r/domain.html.markdown#argument-reference